lib/services/workflow-api-service.js

```diff
@@ -96,11 +96,8 @@ function workflowApiServiceFactory(
                         definition, configuration.options, context, configuration.domain, true);
             });
         })
-        .tap(function(graph) {
-            var progress = GraphProgress.create(graph, 'Graph "' + graph.name + '" started');
-            return eventsProtocol.publishProgressEvent(graph.instanceId,
-                                                       progress.getProgressEventData());
-        })
+        .tap(taskMessenger.publishGraphStarted)
+        .tap(graphProgressService.publishGraphStarted)
```

Comment:
Why not move these two kinds of events into on-taskgraph, on-taskgraph should own how/where to generate the task graph related event.